src/main/prometheus/provider.ts

```diff
@@ -0,0 +1,27 @@
+import { PrometheusHelm } from "./helm"
+import { PrometheusLens } from "./lens"
+import { PrometheusOperator } from "./operator";
+
+export type PrometheusQuery = {
+  [key: string]: string;
+}
+
+export type PrometheusQueryOpts = {
+  [key: string]: string | any;
+};
+
+export interface PrometheusProvider {
+  getQueries(opts: PrometheusQueryOpts): PrometheusQuery;
+}
+
+export class PrometheusProviderFactory {
+  static createProvider(type: string): PrometheusProvider {
+    if (type == "helm") {
```

Comment:
Instead of hard coding id's, each provider could register themselves with an id . Factory could then just find a match from the provider array.

Reply:
I refactored this so that `PrometheusProviderFactory` -> `PrometheusProviderRegistry` and provider objects are added to registry. No need to create a new object on every request.